include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -70,125 +95,302 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
 
+        /** @brief Evaluates non-zero B-spline values and \f$n\f$ derivatives at a given coordinate
+         *
+         * The values and derivatives are computed for every B-spline with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are evaluated and derivated.
+         * A spline approximation of a derivative at coordinate x is a linear
+         * combination of those B-spline derivatives weighted with spline coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] derivs The values and \f$n\f$ derivatives of the B-splines evaluated at coordinate x. It has to be a 2D mdspan with (degree+1)*(n+1) elements.
+         * @param[in] x The coordinate where B-spline derivatives are evaluated.
+         * @param[in] n The number of derivatives to evaluate (in addition to the B-spline values themselves).
+         * @return The index of the first B-spline which is evaluated/derivated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type eval_basis_and_n_derivs(
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * The integral of each of the B-splines over their support within the domain on which this basis was defined.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a 1D mdspan of size (nbasis).
+         * @return The values of the integrals.
+         */
         template <class Layout, class MemorySpace2>
-        KOKKOS_INLINE_FUNCTION ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2>
-        integrals(
-                ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2> int_vals) const;
-
+        KOKKOS_INLINE_FUNCTION ddc::
+                ChunkSpan<double, ddc::DiscreteDomain<DDim>, Layout, MemorySpace2>
+                integrals(ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2>
+                                  int_vals) const;
+
+        /** @brief Returns the coordinate of the knot corresponding to the given index.
+         *
+         * Returns the coordinate of the knot corresponding to the given index. The domain
+         * over which the B-splines are defined is comprised of ncells+1 break points however there are a total of
+         * ncells+1+2*degree knots. The additional knots which control the shape of the B-splines near the
+         * boundary are added before and after the break points. The knot index is therefore in the interval [-degree, ncells+degree]
+         *
+         * @param[in] knot_idx Integer identifying index of the knot.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
```

Comment:
For a future MR, I guess these `get_knot*` functions can be simplified. A single function making the conversion between a `DiscreteElement<BSplines>` and `DiscreteElement<Knots>` should be enough.

Reply:
Indeed. This is part of #416

Reply:
I don't think we can convert from a `DiscreteElement<BSplines>` to a `DiscreteElement<Knots>` as there are d+2 knots associated with each bspline but the knot samplings should be used to replace this function by a call to `coordinate`

Reply:
I was thinking of keeping only `get_first_knot` that would return the first knot `DiscreteElement<Knots>` in the support of the b-spline ? The call to `coordinate` would be up to the user ?

Reply:
I think `get_last_support_knot` can also be useful. I agree that it makes sense for these 2 functions to return `DiscreteElement`s. `get_knot` is equivalent to `coordinate` and `get_support_knot_n` is then equivalent to `get_first_support_knot() + DiscreteVector<Knots>(n)`